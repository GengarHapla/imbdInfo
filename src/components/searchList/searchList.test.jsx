import React from "react";
import SearchList from "./searchList";

describe("SearchList component renders correctly", () => {
  it("when no props", () => {
    expect(<SearchList />).toMatchSnapshot();
  });

  it("when props with 1 item", () => {
    const props = {
      results: {
        Search: [{ imdbID: "1" }]
      }
    };

    expect(<SearchList {...props} />).toMatchSnapshot();
  });

  it("when props with 2 items", () => {
    const props = {
      results: {
        Search: [{ imdbID: "1" }, { imdbID: "2" }]
      }
    };

    expect(<SearchList {...props} />).toMatchSnapshot();
  });

  it("when error", () => {
    const props = {
      results: {
        Error: "An error occured"
      }
    };

    expect(<SearchList {...props} />).toMatchSnapshot();
  });
});
