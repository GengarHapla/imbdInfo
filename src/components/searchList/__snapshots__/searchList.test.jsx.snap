// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchList component renders correctly when error 1`] = `
<Unknown
  results={
    Object {
      "Error": "An error occured",
    }
  }
/>
`;

exports[`SearchList component renders correctly when no props 1`] = `<Unknown />`;

exports[`SearchList component renders correctly when props with 1 item 1`] = `
<Unknown
  results={
    Object {
      "Search": Array [
        Object {
          "imdbID": "1",
        },
      ],
    }
  }
/>
`;

exports[`SearchList component renders correctly when props with 2 items 1`] = `
<Unknown
  results={
    Object {
      "Search": Array [
        Object {
          "imdbID": "1",
        },
        Object {
          "imdbID": "2",
        },
      ],
    }
  }
/>
`;
